modules/rostests/apitests/umkm/i386/XState_asm.s

```diff
@@ -0,0 +1,163 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Assembly helpers for extended state tests
+ * COPYRIGHT:   Copyright 2025 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include <asm.inc>
+
+#ifdef _M_IX86
+#define rcx ecx
+#define get_SSE_state @get_SSE_state@4
+#define set_SSE_state @set_SSE_state@4
+#define get_AVX_state @get_AVX_state@4
+#define set_AVX_state @set_AVX_state@4
+#define get_AVX512_state @get_AVX512_state@4
+#define set_AVX512_state @set_AVX512_state@4
+.code
+#else
+.code64
+#endif
+
+// void __fastcall get_SSE_state(__m128i data[16]);
+PUBLIC get_SSE_state
+get_SSE_state:
+    movaps [rcx + 0], xmm0
```

Comment:
I wonder whether having some sort of macro
`#define GET_XMM(i) movaps [rcx + 16*(i)], xmm##i`
and use it as: `GET_XMM(0)`, `GET_XMM(1)` etc. would be of interest.

Reply:
What for? So that I have more stuff to do?